src/UI/Buyer/src/app/services/order/checkout.service.ts

```diff
@@ -217,17 +218,33 @@ export class CheckoutService {
     const suppliers = await Suppliers.List({filters: {'ID': supplierIDs.join("|")}})
     const supplierItems: LineItemGroupSupplier[] = [];
     for(const combo of uniqueSuppliers) {
-      const supplier = suppliers.Items.find(s => s.ID === combo.supplierID)
-      if(combo.supplierID && combo.ShipFromAddressID) {
-        const shipFrom = await SupplierAddresses.Get(combo.supplierID, combo.ShipFromAddressID);
-        supplierItems.push({supplier,shipFrom})
+      if(combo.supplierID === null) { //This handles seller owned products
+        supplierItems.push(this.buildSellerShipmentData(combo.ShipFromAddressID))
       } else {
-        supplierItems.push({supplier, shipFrom: null})
+        const supplier = suppliers.Items.find(s => s.ID === combo.supplierID)
+        if(combo.supplierID && combo.ShipFromAddressID) {
+          const shipFrom = await SupplierAddresses.Get(combo.supplierID, combo.ShipFromAddressID);
+          supplierItems.push({supplier,shipFrom})
+        } else {
+          supplierItems.push({supplier, shipFrom: null})
+        }
       }
     }
     return supplierItems
   }
 
+  buildSellerShipmentData(shipFromAddresID: string): LineItemGroupSupplier {
+    return {
+      supplier: {
+        ID: null,
+        Name: this.appConfig.sellerName || 'Purchasing from Seller'
```

Comment:
is it possible to have two seller owned products from two different sellers? we would still want them consolidated like this, but maybe we would want just the 'purchasing from Seller' if there's multiples (or just for everything). Otherwise it might be deceiving having "Headstart Demo Seller" (for example), if that's not the only seller. 

Or i'm off base an it's only valid to have one sellerName per order. either works for me!

Reply:
You can only have one sseller per org (the seller kind of is the org). So this should be ok. Can still have more seller addresses but I think we are handling that.